src/main/scala/org/apache/spark/sql/pulsar/PulsarHelper.scala

```diff
@@ -122,7 +135,9 @@ private[pulsar] case class PulsarHelper(
     offset.foreach { case (tp, mid) =>
       try {
         val (subscription, _) = extractSubscription(predefinedSubscription, tp)
-        CachedConsumer.getOrCreate(tp, subscription, client).seek(mid)
+        val consumer = CachedConsumer.getOrCreate(tp, subscription, client)
+        if (!consumer.isConnected) consumer.getLastMessageId
```

Comment:
Does the test still pass if you revert this change?

Reply:
No it doesn't

Reply:
We get the following error:
`stream execution thread for [id = 8039d04a-076f-4717-84e4-1c908945529b, runId = 27e2c374-1aba-4475-b82b-57a0e8d16214]: Query [id = 8039d04a-076f-4717-84e4-1c908945529b, runId = 27e2c374-1aba-4475-b82b-57a0e8d16214] terminated with error
java.lang.RuntimeException: Failed to commit cursor for persistent://public/default/topic-9-partition-1
	at org.apache.spark.sql.pulsar.PulsarHelper.$anonfun$commitCursorToOffset$1(PulsarHelper.scala:144) ~[classes/:?]
	at org.apache.spark.sql.pulsar.PulsarHelper.$anonfun$commitCursorToOffset$1$adapted(PulsarHelper.scala:135) ~[classes/:?]
	at scala.collection.immutable.Map$Map2.foreach(Map.scala:273) ~[scala-library-2.12.17.jar:?]
	at org.apache.spark.sql.pulsar.PulsarHelper.commitCursorToOffset(PulsarHelper.scala:135) ~[classes/:?]
	at org.apache.spark.sql.pulsar.PulsarSource.commit(PulsarSource.scala:193) ~[classes/:?]
	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$commitSources$1(MicroBatchExecution.scala:564) ~[spark-sql_2.12-3.4.0.jar:3.4.0]
	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$commitSources$1$adapted(MicroBatchExecution.scala:563) ~[spark-sql_2.12-3.4.0.jar:3.4.0]
`